cli/azd/cmd/infra_create.go

```diff
@@ -259,7 +259,31 @@ func (ica *infraCreateAction) Run(ctx context.Context, cmd *cobra.Command, args
 		return fmt.Errorf("deployment failed: %w", err)
 	}
 
+	dotnetCli := tools.NewDotNetCli()
+	if err := tools.EnsureInstalled(ctx, dotnetCli); err != nil {
+		return err
+	}
+
+	for _, svc := range proj.Services {
+		if svc.Language == "dotnet" {
+			err := dotnetCli.InitializeSecret(ctx, svc.Path())
+			if err != nil {
+				return err
+			}
```

Comment:
We should avoid adding any special case checks at this point in the codebase.  This check should be done within `dotnet` `FrameworkSerivce` directly.